server/controllers/user.ts

```diff
@@ -124,3 +131,24 @@ export function processLogoutPage(req:Request, res: Response, next: NextFunction
 
     res.redirect("/login");
 }
+
+export function processEditPage(req: Request, res: Response, next: NextFunction): void {
+    const id = req.params.id;
+
+    const updatedUser = new User
+    ({
+        "_id": id,
+        "username": req.body.username,
+        "emailAddress": req.body.email,
+        "contactNumber": req.body.contactnumber
+    });
+    User.updateOne({_id: id}, updatedUser, {}, (err) =>{
+
+        if(err)
+        {
+            console.error(err);
+            res.end(err);
```

Comment:
There are 2 problems with this.

1. The route handler is not finished, so the `res.redirect` is still called in case there is an error. You would want to use `return` to end the route handler function early to prevent that.
2. This sends a short description about the error to the user. By that, you're handling the error all by yourself at the place. But, we already have an error handler in place. You'd want to pass the error to that. To do so, you want to do `next(err)`.

In conclusion, what I suggest is like this:
```ts
if (err) {
    return next(err);
}
```